scene/animation/tween.cpp

```diff
@@ -709,10 +709,10 @@ void Tween::_tween_process(float p_delta) {
 		InterpolateData &data = E->get();
 
 		// Track if we hit one that isn't finished yet
-		all_finished = all_finished && data.finish;
+		all_finished = all_finished && (!data.active || data.finish);
 
 		// Is the data not active or already finished? No need to go any further
-		if (!data.active || data.finish) {
+		if (all_finished) {
```

Comment:
That's better, but still incorrect. This will now mean that if any previous Tween was unfinished, `all_finished` will become and stay `false` (which is good), but subsequent Tweens which are actually finished won't use the early `continue`, leading to unnecessary computation.

Here's the right fix:
```diff
diff --git a/scene/animation/tween.cpp b/scene/animation/tween.cpp
index 854db5fee2..d45fc7dd07 100644
--- a/scene/animation/tween.cpp
+++ b/scene/animation/tween.cpp
@@ -709,7 +709,7 @@ void Tween::_tween_process(float p_delta) {
                InterpolateData &data = E->get();
 
                // Track if we hit one that isn't finished yet
-               all_finished = all_finished && data.finish;
+               all_finished = all_finished && (!data.active || data.finish);
 
                // Is the data not active or already finished? No need to go any further
                if (!data.active || data.finish) {
```